cores/arduino/stm32/interrupt.cpp

```diff
@@ -62,8 +62,9 @@
 
 /*As we can have only one interrupt/pin id, don't need to get the port info*/
 typedef struct {
-  uint32_t irqnb;
-  void (*callback)(void);
+  IRQn_Type irqnb;
+  // void (*callback)(void);
```

Comment:
this one as well can be removed

Reply:
I had to change the type of irqnb because the compiler give me the error that can't convert uint32_t to IRQn_Type. 
From what i know the default type of an enum is INT

Reply:
hmm - when did the compiler error appeared ? only after some your changes ?

Reply:
I haven't tried without my changes... i think not..
I'm in OSX, can be a problem of arm_gcc under osx?
Now i can't do tests....

Reply:
I'll let other reviewers comment as well whether this change is problematic or not

Reply:
Ok np, this is the right choice!

Reply:
Here the error:

```
~/Library/Arduino15/packages/STM32/hardware/stm32/2017.11.28/cores/arduino/stm32/interrupt.cpp: In function 'void stm32_interrupt_enable(GPIO_TypeDef*, uint16_t, callback_function_t, uint32_t)':
~/Library/Arduino15/packages/STM32/hardware/stm32/2017.11.28/cores/arduino/stm32/interrupt.cpp:217:42: error: invalid conversion from 'uint32_t {aka long unsigned int}' to 'IRQn_Type' [-fpermissive]
   HAL_NVIC_SetPriority(gpio_irq_conf[id].irqnb, 0x06, 0);
                        ~~~~~~~~~~~~~~~~~~^~~~~
In file included from ~/Library/Arduino15/packages/STM32/hardware/stm32/2017.11.28/variants/NUCLEO_F429ZI/stm32f4xx_hal_conf.h:262:0,
                 from ~/Library/Arduino15/packages/STM32/hardware/stm32/2017.11.28/system/Drivers/STM32F4xx_HAL_Driver/Inc/stm32f4xx_hal.h:46,
                 from ~/Library/Arduino15/packages/STM32/hardware/stm32/2017.11.28/system/Drivers/CMSIS/Device/ST/STM32F4xx/Include/stm32f4xx.h:250,
                 from ~/Library/Arduino15/packages/STM32/hardware/stm32/2017.11.28/cores/arduino/stm32/stm32_def.h:36,
                 from ~/Library/Arduino15/packages/STM32/hardware/stm32/2017.11.28/cores/arduino/stm32/interrupt.cpp:49:
~/Library/Arduino15/packages/STM32/hardware/stm32/2017.11.28/system/Drivers/STM32F4xx_HAL_Driver/Inc/stm32f4xx_hal_cortex.h:279:6: note:   initializing argument 1 of 'void HAL_NVIC_SetPriority(IRQn_Type, uint32_t, uint32_t)'
 void HAL_NVIC_SetPriority(IRQn_Type IRQn, uint32_t PreemptPriority, uint32_t SubPriority);

      ^~~~~~~~~~~~~~~~~~~~
~/Library/Arduino15/packages/STM32/hardware/stm32/2017.11.28/cores/arduino/stm32/interrupt.cpp:218:40: error: invalid conversion from 'uint32_t {aka long unsigned int}' to 'IRQn_Type' [-fpermissive]
   HAL_NVIC_EnableIRQ(gpio_irq_conf[id].irqnb);
                      ~~~~~~~~~~~~~~~~~~^~~~~
In file included from ~/Library/Arduino15/packages/STM32/hardware/stm32/2017.11.28/variants/NUCLEO_F429ZI/stm32f4xx_hal_conf.h:262:0,
                 from ~/Library/Arduino15/packages/STM32/hardware/stm32/2017.11.28/system/Drivers/STM32F4xx_HAL_Driver/Inc/stm32f4xx_hal.h:46,
                 from ~/Library/Arduino15/packages/STM32/hardware/stm32/2017.11.28/system/Drivers/CMSIS/Device/ST/STM32F4xx/Include/stm32f4xx.h:250,
                 from ~/Library/Arduino15/packages/STM32/hardware/stm32/2017.11.28/cores/arduino/stm32/stm32_def.h:36,
                 from ~/Library/Arduino15/packages/STM32/hardware/stm32/2017.11.28/cores/arduino/stm32/interrupt.cpp:49:
~/Library/Arduino15/packages/STM32/hardware/stm32/2017.11.28/system/Drivers/STM32F4xx_HAL_Driver/Inc/stm32f4xx_hal_cortex.h:280:6: note:   initializing argument 1 of 'void HAL_NVIC_EnableIRQ(IRQn_Type)'
 void HAL_NVIC_EnableIRQ(IRQn_Type IRQn);

      ^~~~~~~~~~~~~~~~~~
~/Library/Arduino15/packages/STM32/hardware/stm32/2017.11.28/cores/arduino/stm32/interrupt.cpp: In function 'void stm32_interrupt_disable(GPIO_TypeDef*, uint16_t)':
~/Library/Arduino15/packages/STM32/hardware/stm32/2017.11.28/cores/arduino/stm32/interrupt.cpp:254:41: error: invalid conversion from 'uint32_t {aka long unsigned int}' to 'IRQn_Type' [-fpermissive]
   HAL_NVIC_DisableIRQ(gpio_irq_conf[id].irqnb);
                       ~~~~~~~~~~~~~~~~~~^~~~~
In file included from ~/Library/Arduino15/packages/STM32/hardware/stm32/2017.11.28/variants/NUCLEO_F429ZI/stm32f4xx_hal_conf.h:262:0,
                 from ~/Library/Arduino15/packages/STM32/hardware/stm32/2017.11.28/system/Drivers/STM32F4xx_HAL_Driver/Inc/stm32f4xx_hal.h:46,
                 from ~/Library/Arduino15/packages/STM32/hardware/stm32/2017.11.28/system/Drivers/CMSIS/Device/ST/STM32F4xx/Include/stm32f4xx.h:250,
                 from ~/Library/Arduino15/packages/STM32/hardware/stm32/2017.11.28/cores/arduino/stm32/stm32_def.h:36,
                 from ~/Library/Arduino15/packages/STM32/hardware/stm32/2017.11.28/cores/arduino/stm32/interrupt.cpp:49:
~/Library/Arduino15/packages/STM32/hardware/stm32/2017.11.28/system/Drivers/STM32F4xx_HAL_Driver/Inc/stm32f4xx_hal_cortex.h:281:6: note:   initializing argument 1 of 'void HAL_NVIC_DisableIRQ(IRQn_Type)'
 void HAL_NVIC_DisableIRQ(IRQn_Type IRQn);

      ^~~~~~~~~~~~~~~~~~~
Using library LwIP at version 2.0.3 in folder: ~/Documents/Arduino/libraries/LwIP 
Using library STM32Ethernet at version 1.0.2 in folder: ~/Documents/Arduino/libraries/STM32Ethernet 
exit status 1
Error compiling for board Nucleo-144.
```

Reply:
Makes sense to me to use IRQn_Type but I don't understand why this was not visible before. 
@fprwi6labs @fpistm  can you please cross-check if this is fine with you ?

Reply:
Yes, as file is now cpp. Compiler do not allow it.
Fix is ok.